app/Http/Controllers/OvertimeRequestController.php

```diff
@@ -0,0 +1,214 @@
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Inertia\Response;
+use Toby\Actions\OvertimeRequest\AcceptAsTechnicalAction;
+use Toby\Actions\OvertimeRequest\CancelAction;
+use Toby\Actions\OvertimeRequest\CreateAction;
+use Toby\Actions\OvertimeRequest\RejectAction;
+use Toby\Actions\OvertimeRequest\SettleAction;
+use Toby\Domain\OvertimeRequestStatesRetriever;
+use Toby\Enums\SettlementType;
+use Toby\Helpers\YearPeriodRetriever;
+use Toby\Http\Requests\OvertimeRequestRequest;
+use Toby\Http\Resources\OvertimeRequestActivityResource;
+use Toby\Http\Resources\OvertimeRequestResource;
+use Toby\Http\Resources\SimpleUserResource;
+use Toby\Models\OvertimeRequest;
+use Toby\Models\User;
+
+class OvertimeRequestController extends Controller
+{
+    public function index(Request $request, YearPeriodRetriever $yearPeriodRetriever): Response|RedirectResponse
+    {
+        $this->authorize("canUseOvertimeRequestFunctionality", $request->user());
+
+        $status = $request->get("status", "all");
+
+        $overtimeRequests = $request->user()
+            ->overtimeRequests()
+            ->with(["user.permissions", "user.profile"])
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->latest()
+            ->states(OvertimeRequestStatesRetriever::filterByStatusGroup($status, $request->user()))
+            ->paginate();
+
+        $pending = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::pendingStates())
+            ->count();
+
+        $success = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::successStates())
+            ->count();
+
+        $failed = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::failedStates())
+            ->count();
+
+        $settled = $request->user()
+            ->overtimeRequests()
+            ->whereBelongsTo($yearPeriodRetriever->selected())
+            ->states(OvertimeRequestStatesRetriever::settledStates())
+            ->count();
+
+        return inertia("OvertimeRequest/Index", [
+            "requests" => OvertimeRequestResource::collection($overtimeRequests),
+            "stats" => [
+                "all" => $pending + $success + $failed + $settled,
+                "pending" => $pending,
+                "success" => $success,
+                "failed" => $failed,
+                "settled" => $settled,
+            ],
+            "filters" => [
+                "status" => $status,
+            ],
+        ]);
+    }
+
+    public function indexForApprovers(
+        Request $request,
+        YearPeriodRetriever $yearPeriodRetriever,
+    ): RedirectResponse|Response {
```

Comment:
Maybe throwing an exception with redirect would be better?

Reply:
added `abort(403)`